source/common/http/utility.cc

```diff
@@ -747,5 +749,49 @@ std::string Utility::PercentEncoding::decode(absl::string_view encoded) {
   return decoded;
 }
 
+Utility::AuthorityAttributes Utility::parseAuthority(absl::string_view host) {
+  // First try to see if there is a port included. This also checks to see that there is not a ']'
+  // as the last character which is indicative of an IPv6 address without a port. This is a best
+  // effort attempt.
+  const auto colon_pos = host.rfind(':');
+  absl::string_view host_to_resolve = host;
+  absl::optional<uint16_t> port;
+  if (colon_pos != absl::string_view::npos && host_to_resolve.back() != ']') {
```

Comment:
What happens if colon_pos == 0, i.e. someone gives  ":1232" in the authority header?

Reply:
I think that's safe; that's still not `npos`. But, this does raise an interesting question, how well formed are we guaranteed the IPv6 addresses look? Are our codecs verifying this?

Reply:
SGTM. I think the issue could be passing empty hostname to the DNS resolver and unexpected things happening.